openfeature/multi/comparison_strategy.go

```diff
@@ -0,0 +1,249 @@
+package multi
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"slices"
+	"strings"
+
+	of "github.com/open-feature/go-sdk/openfeature"
+	"golang.org/x/sync/errgroup"
+)
+
+// ErrAggregationNotAllowed is an error returned if [of.FeatureProvider.ObjectEvaluation] is called using the [StrategyComparison]
+// strategy without a custom Comparator function configured when response objects are not comparable.
+var ErrAggregationNotAllowed error = errors.New(errAggregationNotAllowedText)
+
+// Comparator function used for comparing results of [of.FeatureProvider.ObjectEvaluation]. This is required if returned
+// results are not comparable.
+type Comparator func(values []any) bool
+
+// newComparisonStrategy creates a new instance of ComparisonStrategy. The fallback provider specified is called when
```

Comment:
This first sentence seems misleading as there is no identifier named `ComparisonStrategy`. I know this isn't exported to users, but I'd still want this to be more accurate. Maybe something like this? Let me know what you think.
> newComparisonStrategy returns a [StrategyComparison] strategy function.